internal/plugincommon/discovery/processor.go

```diff
@@ -182,28 +180,55 @@ func aggregateObservations(
 	return obs
 }
 
-// Outcome comes to consensus on the contract addresses and updates the chainreader. It doesn't actually
-// return an Outcome.
+func getRouterConsensus(
```

Comment:
This function could use some specific attention.

TODO: add a comment describing the approach used here.

Reply:
Can we add a unit test suite for this func?